time start_stamp
time finish_stamp